src/main/java/mmosii/bookstore/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -44,13 +45,13 @@ public CartItemDto addCartItem(CartItemRequestDto requestDto) {
         return cartItemMapper.toDto(cartItemRepository.save(cartItem));
     }
 
+    @Transactional
     @Override
     public CartItemDto updateCartItem(Long itemId, CartItemUpdateDto updateDto) {
         ShoppingCart shoppingCart = getUserCart();
         CartItem cartItem = findCartItem(
                 itemId, shoppingCart.getId(), shoppingCart.getUser().getId());
         cartItem.setQuantity(updateDto.quantity());
-        cartItemRepository.save(cartItem);
         return cartItemMapper.toDto(cartItem);
```

Comment:
Why do you remove save? How should it work?

Reply:
with @Transactional on it it works without save)
https://stackoverflow.com/questions/46708063/springboot-jpa-need-no-save-on-transactional